atomistics/calculators/sphinxdft.py

```diff
@@ -0,0 +1,160 @@
+import os
+
+from ase.atoms import Atoms
+from sphinx_parser.ase import get_structure_group
+from sphinx_parser.input import sphinx
+from sphinx_parser.output import collect_energy_dat, collect_eval_forces
+from sphinx_parser.potential import get_paw_from_structure
+from sphinx_parser.toolkit import to_sphinx
+
+from atomistics.calculators.interface import get_quantities_from_tasks
+from atomistics.calculators.wrapper import as_task_dict_evaluator
+from atomistics.shared.output import OutputStatic
+
+
+def _write_input(
+    structure: Atoms,
+    working_directory: str,
+    maxSteps: int = 100,
+    eCut: float = 25,
+ def _write_input(
+     structure: Atoms,
+     working_directory: str,
+     maxSteps: int = 100,
+     eCut: float = 25,
+     kpoint_coords: list[float, float, float] | None = None,
+     kpoint_folding: list[int, int, int] | None = None,
+ ):
+     if kpoint_coords is None:
+         kpoint_coords = [0.5, 0.5, 0.5]
+     if kpoint_folding is None:
+         kpoint_folding = [4, 4, 4]
+     # Rest of the function body...
+):
+    struct_group, spin_lst = get_structure_group(structure)
+    main_group = sphinx.main.create(
+        scfDiag=sphinx.main.scfDiag.create(maxSteps=maxSteps, blockCCG={}),
+        evalForces=sphinx.main.evalForces.create(file="relaxHist.sx"),
+    )
+    pawPot_group = get_paw_from_structure(structure)
+    basis_group = sphinx.basis.create(
+        eCut=eCut,
+        kPoint=sphinx.basis.kPoint.create(coords=kpoint_coords),
+        folding=kpoint_folding,
+    )
+    paw_group = sphinx.PAWHamiltonian.create(xc=1, spinPolarized=False, ekt=0.2)
+    initial_guess_group = sphinx.initialGuess.create(
+        waves=sphinx.initialGuess.waves.create(
+            lcao=sphinx.initialGuess.waves.lcao.create()
+        ),
+        rho=sphinx.initialGuess.rho.create(atomicOrbitals=True),
+    )
+    input_sx = sphinx.create(
+        pawPot=pawPot_group,
+        structure=struct_group,
+        main=main_group,
+        basis=basis_group,
+        PAWHamiltonian=paw_group,
+        initialGuess=initial_guess_group,
+    )
+    with open(os.path.join(working_directory, "input.sx"), "w") as f:
+        f.write(to_sphinx(input_sx))
+
+
+class OutputParser:
+    def __init__(self, working_directory: str, structure: Atoms):
+        self._working_directory = working_directory
+        self._structure = structure
+
+    def get_energy(self):
+        return collect_energy_dat(os.path.join(self._working_directory, "energy.dat"))[
+            "scf_energy_int"
+        ][-1][-1]
+
+    def get_forces(self):
+        return collect_eval_forces(
+            os.path.join(self._working_directory, "relaxHist.sx")
+        )["forces"][-1]
```

Comment:
You have to reorder the atoms using `sphinx_parser.ase.id_spx_to_ase`. Probably it's enough to just add `[id_spx_to_ase(self._structure)]`

Reply:
Thanks for the suggestion